src/metrics.jl

```diff
@@ -214,6 +250,28 @@ eval_end(::Euclidean, s) = sqrt(s)
 euclidean(a::AbstractArray, b::AbstractArray) = evaluate(Euclidean(), a, b)
 euclidean(a::Number, b::Number) = evaluate(Euclidean(), a, b)
 
+# PeriodicEuclidean
+Base.eltype(d::PeriodicEuclidean) = eltype(d.periods)
+@inline function eval_start(d::PeriodicEuclidean, a::AbstractArray, b::AbstractArray)
+    (zero(result_type(d, a, b)), d.periods)
+end
+@inline eval_op(::PeriodicEuclidean, ai, bi) = abs(ai - bi)
+@inline function eval_reduce(::PeriodicEuclidean, s1, s2)
+    li = first(s1[2])
+    d = mod(s2, li)
+    d = min(d, li - d)
+    (s1[1] + abs2(d), Base.tail(s1[2]))
+end
+@inline eval_end(::PeriodicEuclidean, s) = sqrt(s[1])
+function evaluate(dist::PeriodicEuclidean, a::T, b::T) where {T <: Number}
```

Comment:
`Real` was used above instead of `Number`. Any reason to diverge?

Reply:
No, just oversight.